File: src/main/java/Lec08/Lec08.java
package Lec08;

public class Lec08 {
    
    public static void main(String[] args) {
        String[] arr = new String[]{"A", "B", "C"};
        printAll(arr);
        
        printAll("D", "E", "F");
    }
    
    public int max(int a, int b) {
        if (a > b) {
            return a;
        }
        return b;
    }
    
    public void repeat(String str, int num, boolean useNewLine) {
        for (int i = 1; i <= num; i++) {
            if (useNewLine) {
                System.out.println(str);
            } else {
                System.out.print(str);
            }
        }
    }
    
    public void repeat(String str, int num) {
        repeat(str, num, true);
    }
    
    public void repeat(String str) {
        repeat(str, 3, true);
    }
    
    public static void printAll(String... strings) {
        for (String str : strings) {
            System.out.println(str);
        }
    }
}

File: src/main/kotlin/Lec08/Lec08.kt
import Lec08.Lec08

fun main() {
    repeat("Hello", 3, true)
    repeat("Hi")
    repeat("Bye", useNewLine = false)

    printNameAndGender(name = "developer", gender = "MALE")

    val arr = arrayOf("A", "B", "C")
    printAll(*arr)
    printAll("D", "E", "F")
}

fun max(a: Int, b: Int) = if (a > b) a else b

fun repeat(
    str: String,
    num: Int = 3,
    useNewLine: Boolean = true
) {
    for (i in 1..num) {
        if (useNewLine) {
            println(str)
        } else {
            print(str)
        }
    }
}

fun printNameAndGender(name: String, gender: String) {
    println(name)
    println(gender)
}

fun printAll(vararg strings: String) {
    for (str in strings) {
        println(str)
    }
}
